router/core/graphql_prehandler.go

```diff
@@ -260,340 +273,304 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 			readOperationBodySpan.End()
 		}
 
-		/**
-		 * Load and Parse the operation. Optionally, we will fetch the persisted operation from the CDN
-		 */
-
-		operationKit, err := h.operationProcessor.NewKit(body, files)
+		opContext, err := h.handleOperation(r, buf, &httpOperation{
+			requestLogger:    requestLogger,
+			attributes:       commonAttributes,
+			clientInfo:       clientInfo,
+			routerSpan:       routerSpan,
+			operationMetrics: metrics,
+			traceOptions:     traceOptions,
+			traceTimings:     traceTimings,
+			executionOptions: executionOptions,
+			files:            files,
+			body:             body,
+		})
 		if err != nil {
 			finalErr = err
-
-			// Mark the root span of the router as failed, so we can easily identify failed requests
-			rtrace.AttachErrToSpan(routerSpan, err)
-
-			writeOperationError(r, w, requestLogger, err)
-			h.releaseBodyReadBuffer(buf)
-			return
-		}
-
-		if err := operationKit.UnmarshalOperation(); err != nil {
-			// the kit must be freed before we're doing io operations
-			// the kit is bound to the number of CPUs, and we must not hold onto it while doing IO operations
-			operationKit.Free()
-			operationKit = nil
-			finalErr = err
-
 			// Mark the root span of the router as failed, so we can easily identify failed requests
 			rtrace.AttachErrToSpan(routerSpan, err)
 
 			writeOperationError(r, w, requestLogger, err)
-			h.releaseBodyReadBuffer(buf)
```

Comment:
Do we still release the buf like before?

Reply:
Yes, same behaviour in `handleOperation`